File: Client/src/type/Cart/CartResponse.ts
// src/type/cart/CartResponse.ts

export interface CartItemDto {
  cartItemId: string;
  productVariantId: string;
  productId: string;
  productName: string;
  colorCode: string;
  colorName: string;
  sizeId: string;
  sizeName: string;
  img: string;
  price: number;
  quantity: number;
  subtotal: number;
}

export interface CartResponse {
  id: string;
  items: CartItemDto[];
  totalAmount: number;
}

// src/type/cart/AddToCartRequest.ts
export interface AddToCartRequest {
  productVariantId: string;
  quantity: number;
}

File: src/main/java/com/example/Server/dto/response/cart/CartResponse.java
package com.example.Server.dto.response.cart;

import lombok.*;
import java.util.List;

@Data @NoArgsConstructor @AllArgsConstructor
public class CartResponse {
    private String id;
    private List<CartItemDto> items;
    private Double totalAmount;

    @Data @NoArgsConstructor @AllArgsConstructor
    public static class CartItemDto {
        private String cartItemId;
        private String productVariantId;
        private String productId;
        private String productName;
        private String colorCode;
        private String colorName;
        private String sizeId;
        private String sizeName;
        private String img;
        private Double price;
        private Integer quantity;
        private Double subtotal;
    }
}
